MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -206,5 +203,98 @@ private int CalculateBytesPerPixel()
                     return -1;
             }
         }
+
+        private void GetColorData(Texture2D texture2D)
```

Comment:
Agreed, it's too general to be confined to `PngWriter`.  `Texture2D` seems like a good place for it eventually.